app/controllers/users/omniauth_controller.rb

```diff
@@ -24,7 +24,9 @@ def callback
     def allow_unregistered_government_user(received_email)
       return if @user
       allowed_tlds = Figaro.env.auto_account_creation_tlds.split(',')
-      return unless allowed_tlds.include?(received_email[-4..-1])
+      return if allowed_tlds.filter do |tld|
+        /(#{Regexp.escape(tld)})$/.match?(received_email)
```

Comment:
User `\Z` as an anchor? it's more complete than `$` because it matches end of input and not end of line (in case somebody sneaks in a multi-line email address)

Reply:
Also maybe instead of `.filter {}.empty?` use `.none?`

```ruby
return if allowed_tlds.none? |tld|
  # etc
end
```